Source/AwsCommonRuntimeKit/crt/Future.swift

```diff
@@ -0,0 +1,61 @@
+//  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//  SPDX-License-Identifier: Apache-2.0.
+
+import Dispatch
+
+public class Future<Value> {
+    public typealias FutureResult = Result<Value, Error>
+
+    var _value: FutureResult?
```

Comment:
is value thread safe? should it at least be atomic?

Reply:
tried to address this, completely valid point